### ISSUE TITLE
RealmChangeSet contains all or most objects

### ISSUE THREAD

**Issue by grangej** [realm/realm-cocoa#6032]
**Bugs**: I seem to be running into a problem with the realm change set, I am observing a collection of objects and it keeps telling me that all or most of the objects are updated in the collection, even when I only updated one object. 

In the example of the single update in a collection of 40 items It return 12 updates. When I append 20 items to the end (older items by the sort) the changeSet returns that the entire collection was updated. These behaviors of course defeat the purpose of watching for changes :(

## Goals
Need to know inserts, deletes, changes for tableView refresh

## Expected Results
Changes are returned only for objects that I have modified or that might have been re-ordered. 

## Actual Results
Changes appear to be returned for many more objects than what are updated or moved.

## Steps to Reproduce
Observing a change set of objects that are sorted. 

## Code Sample
`        let notifications = realm.objects(ReachNotification.self).filter("owner = %@", customer).sorted(byKeyPath: "creationTime", ascending: false)

        //self.notifications = Observable.array(from: notifications)
        notificationChangeSet = Observable.arrayWithChangeset(from: notifications)
`
## Version of Realm and Tooling

Realm framework version: 3.75

Realm Object Server version: N/A

Xcode version: 10.1

iOS/OSX version: 11

Dependency manager + version: Cocoapod

**Comment by Bogsey** [comment]
I have something similar. A set of results that are filtered by one property and sorted by date. If I change a property (not the sorted date property or the filter property) on one of the results then the changeset includes modifications for all results however if I look at fine grained notifications for the objects then there are no changes.

**Comment by grangej** [comment]
Actually I solved this, the problem was that one of the related objects was being re-created using realm.add(object: update:) this caused the root object to trigger an update. If instead the repeated object was returned as the existing object this no longer happened. In addition I created some helper methods for inserting objects based on dictionary's that would return existing objects if the insert dictionary was the same.
